### ISSUE TITLE
Unable to add host - healthcheck and ipsec are failing

### ISSUE THREAD

**Issue by vikas027** [rancher/rancher#7789]
**Rancher Versions:**
Server: 1.4
healthcheck: 0.2.3
ipsec: 0.8.7
network-services: 0.4.5
scheduler:0.6.3

**Docker Version:**
Tested on version 1.12.3 and 1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**
Local VMs on VMware Fusion
```sh
CentOS Linux release 7.3.1611 (Core)
```

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
#### On Server
```sh
$ docker run --name rancher-server -dit --restart=unless-stopped -p 8080:8080 rancher/server
```

#### Add Agent / Hosts
INFRASTRUCTURE -> Hosts -> Add Host

```sh
$ docker run -d --privileged -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/rancher:/var/lib/rancher rancher/agent:v1.2.0 http://192.168.1.1:8080/v1/scripts/26908E5A76F3F13B88C9:1483142400000:99fvX5RQdHUO53juZwILZrHUc
```

```sh
root@box86:~# docker ps -a
CONTAINER ID        IMAGE                            COMMAND                  CREATED              STATUS                     PORTS               NAMES
0c9fea68ed2f        rancher/scheduler:v0.6.3         "/.r/r /rancher-entry"   34 seconds ago       Up 33 seconds                                  r-scheduler-scheduler-1-a1b47361
726f05c7ce03        rancher/net:holder               "/.r/r /rancher-entry"   47 seconds ago       Up 46 seconds                                  r-ipsec-ipsec-1-c61e44ea
b724748abc19        rancher/net:v0.8.7               "/rancher-entrypoint."   48 seconds ago       Up 48 seconds                                  r-ipsec-ipsec-cni-driver-1-2a81afd3
1f2468a1f2a7        rancher/healthcheck:v0.2.3       "/.r/r /rancher-entry"   About a minute ago   Up About a minute                              r-healthcheck-healthcheck-1-0013d5ca
1a1facdd08b9        rancher/dns:v0.13.3              "/rancher-entrypoint."   5 minutes ago        Up 5 minutes                                   r-network-services-metadata-dns-1-f3c09910
37e6b1a537a3        rancher/metadata:v0.7.0          "/rancher-entrypoint."   5 minutes ago        Up 5 minutes                                   r-network-services-metadata-1-77e05f83
2a4015bdae57        rancher/network-manager:v0.4.5   "/rancher-entrypoint."   5 minutes ago        Up 5 minutes                                   r-network-services-network-manager-1-216504a0
334827ff2380        rancher/agent:v1.2.0             "/run.sh run"            5 minutes ago        Up 5 minutes                                   rancher-agent
67df21a29231        rancher/agent:v1.2.0             "/run.sh http://192.1"   5 minutes ago        Exited (0) 5 minutes ago                       agitated_cray
root@box86:~#
```

**Results:**
![2017-02-13_00-26-02](https://cloud.githubusercontent.com/assets/6998650/22862458/15104de4-f183-11e6-8d8a-cc1365a70ba0.png)
There are lots of timeout  in docker logs
```sh
2017-02-12 13:26:26,276 ERROR [5e32b290-5771-4406-9518-182766cc22de:134] [service:2] [service.activate] [] [ecutorService-2] [c.p.e.p.i.DefaultProcessInstanceImpl] Expected state running but got error: Timeout getting IP address
2017-02-12 13:26:26,280 ERROR [:] [] [] [] [ecutorService-2] [.e.s.i.ProcessInstanceDispatcherImpl] Expected state running but got error: Timeout getting IP address
```

**Comment by Dmitry1987** [comment]
Did you allow needed UDP ports between slave and master nodes?
as in 
```
Make sure any security groups or firewalls allow traffic:
From and To all other hosts on UDP ports 500 and 4500 (for IPsec networking)
```
looks like connection issues ..   you can also verify if UDP allowed [like shown here](https://www.digitalocean.com/community/tutorials/how-to-use-netcat-to-establish-and-test-tcp-and-udp-connections-on-a-vps)

hope it helps

**Comment by aemneina** [comment]
@vikas027 try these things:
- disable selinux 
- make sure you're using the latest kernel available

The rancher server does not need access to the agent nodes over udp. The server doesn't partake in the ipsec network.

**Comment by vikas027** [comment]
@Dmitry1987 @aemneina 

In both rancher server and host, I already had `selinux` and `firewalld/iptables`disabled, also had upgraded to latest kernel available in `CentOS` base repo

```sh
# sestatus
SELinux status:                 disabled
#
# systemctl status firewalld
● firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
     Docs: man:firewalld(1)
#
# uname -r
3.10.0-514.6.1.el7.x86_64
#
```

**Comment by aemneina** [comment]
its okay to disable firewalld, but you need iptables.

**Comment by vikas027** [comment]
@aemneina Yeah, `iptables` managed by Docker itself is working fine.

```sh
# iptables -nvL
Chain INPUT (policy ACCEPT 1271 packets, 196K bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
 3950  473K CATTLE_FORWARD  all  --  *      *       0.0.0.0/0            0.0.0.0/0
 4461  751K DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0
 3817  678K DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0
 3035  617K ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
  644 72358 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0
  782 61470 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0

Chain OUTPUT (policy ACCEPT 1448 packets, 321K bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain CATTLE_FORWARD (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            mark match 0x1068

Chain DOCKER (1 references)
 pkts bytes target     prot opt in     out     source               destination

Chain DOCKER-ISOLATION (1 references)
 pkts bytes target     prot opt in     out     source               destination
 4461  751K RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0
```

**Comment by vikas027** [comment]
@aemneina  @Dmitry1987 I have found a workaround to my problem. 

My VMs in VMware Fusion have two network cards

- NIC 1 - 192.168.1.X  (Host Only Network; networking between VMs)
- NIC 2 - 192.168.181.X (NAT Network; to connect to Internet)

These are my default routes and DNS
```sh
# netstat -rn
Kernel IP routing table
Destination     Gateway         Genmask         Flags   MSS Window  irtt Iface
0.0.0.0         192.168.181.2   0.0.0.0         UG        0 0          0 ens37
172.17.0.0      0.0.0.0         255.255.0.0     U         0 0          0 docker0
192.168.1.0     0.0.0.0         255.255.255.0   U         0 0          0 ens33
192.168.181.0   0.0.0.0         255.255.255.0   U         0 0          0 ens37
#

# cat /etc/resolv.conf
# Generated by NetworkManager
search localdomain test.com
nameserver 192.168.181.2
nameserver 192.168.1.100
#
```

Now, everything starts working if I swap my DNS servers, OR disable my NAT network
```sh
# cat /etc/resolv.conf
search localdomain test.com
nameserver 192.168.1.100
nameserver 192.168.181.2
#
```

I am not sure about this behavior. Networking has never been a problem in umpteen other VMs running on my machine. 

This has something to do with routes on VM, I will fix will fix it sometime soon.

**Comment by vikas027** [comment]
@Dmitry1987 @aemneina 

I had the same issue even after using just one network (NAT) interface, I did the same to remove _complexity_. Looks like Rancher is very particular about the `search domain`.

For me this worked
```sh
root@box71:~# cat /etc/resolv.conf
search localdomain
nameserver 192.168.181.2
root@box71:~#
```
and this didn't
```sh
root@box71:~# cat /etc/resolv.conf
search test.com
nameserver 192.168.181.2
root@box71:~#
```

In either case, my container and hosts are able to reach the internet without any issues as I have enabled `DNS forwarder` enabled.

Anyways, I am glad it is working finally... time to play with the API. :)

Hopefully, I will not face these issues in my prod (AWS) environment.
